core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +88,43 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v32_dict_to_v33_dict(cls, draft_change_list):
+        """Converts draft change list from state version 32 to 33. State
+        version 33 adds showChoicesInShuffledOrder boolean variable to the
+        MultipleChoiceInput interaction.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        multiple_choice_states = []
+        for i, change in enumerate(draft_change_list):
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_INTERACTION_ID):
+                if change.new_value == 'MultipleChoiceInput':
+                    multiple_choice_states.append(change.state_name)
```

Comment:
You still can't do this. The user may change the interaction of a state, and then the state won't be a multiple-choice state any more (or the other way round). Or the state may have already been a multiple-choice state from long ago and you'll miss it.

I think what you need to do is just a single for loop. Once you get to a cust args property that might be relevant, you'll need to check if the *current* interaction ID of the state is correct. One way you could check this is to look at the format of the existing cust args property (is there a way to uniquely identify such properties that belong to MultipleChoiceInput -- do they have a particular structure that the cust args for other interactions do not have)? If that doesn't work, you'll need to do something more complex, like figuring out what the interactions for each state are at the beginning and then updating that dict as you iterate through the draft changes.

Reply:
Done.